File: go/vulnerable2.go
/**
 * Additional vulnerable Go code samples for SAST testing
 * Author: Test Suite
 * Created: 2025-01-21
 */

package main

import (
	"database/sql"
	"fmt"
	"os/exec"
)

// SQL Injection
func searchUsers(searchTerm string) (*sql.Rows, error) {
	query := fmt.Sprintf("SELECT * FROM users WHERE name LIKE '%%%s%%'", searchTerm)
	return db.Query(query)
}

// Hardcoded secrets
var dbConfig = map[string]string{
	"host":     "localhost",
	"user":     "admin",
	"password": "admin123",
	"database": "mydb",
}

// Command Injection
func runCommand(cmd string) error {
	return exec.Command("sh", "-c", cmd).Run()
}

File: javascript/vulnerable2.js
/**
 * @fileoverview Additional vulnerable JavaScript code samples for SAST testing
 * @author Test Suite
 * @created 2025-01-21
 */

// SQL Injection with template literals
function searchUsers(searchTerm) {
  const query = `SELECT * FROM users WHERE name LIKE '%${searchTerm}%'`;
  return db.query(query);
}

// Hardcoded credentials
const dbConfig = {
  host: 'localhost',
  user: 'admin',
  password: 'password123',
  database: 'mydb'
};

// XSS - React dangerouslySetInnerHTML
function UserProfile({ bio }) {
  return <div dangerouslySetInnerHTML={{ __html: bio }} />;
}

// Command Injection with child_process
const { spawn } = require('child_process');
function runScript(scriptName) {
  spawn('sh', ['-c', scriptName]);
}

